src/generation.rs

```diff
@@ -1259,6 +1272,16 @@ impl GenerationScope {
                             deser_code.content.line(&format!("{}{}{}", before_after.before_str(false), final_expr(config.final_exprs, None), before_after.after_str(false)));
                         }
                     },
+                    FixedValue::Float(x) => {
+                        deser_code.content.line(&format!("let {}_value = {}.float()?;", config.var_name, deserializer_name));
+                        let mut compare_block = Block::new(&format!("if {}_value != {}", config.var_name, x));
+                        compare_block.line(format!("return Err(DeserializeFailure::FixedValueMismatch{{ found: Key::Float({}_value), expected: Key::Float({}) }}.into());", config.var_name, x));
+                        deser_code.content.push_block(compare_block);
+                        if CLI_ARGS.preserve_encodings {
```

Comment:
Could you also add a test for preserving encodings?

Reply:
Having looked through this more in depth now, I think I have confirmed my initial suspicion that it doesn't support it for floats. In CBOR you can have multiple sizes for floats: `float16` (major type #`7.25`), `float32` (`#7.26`), `float64` (`#7.27`) (+ type choices on those). The update to `cbor_event` only seems to have `write_f64()` (the one using `Special::Float` only uses the `write_f64`). The API we use for `uint`/`nint`/etc are the `Deserializer::*_sz()`/`Serializer::write_*_sz()` ones which tell us how many bytes were used in the deserialization, and let us specify it for serialization.

If we're not going to support preserving encodings for floats then we should at least make it panic if you pass in a .cddl with floats and have `--preserve-encodings=true`. This configuration is very important for us, but we also don't (currently) need float support with encodings preserving (i.e. where specific CBOR formats are important for cryptographic stuff) generally as floats tend to be avoided in IOHKs' protocols.

If we want to instead try and support it we'd need to add it to `cbor_event` similar to primetype/cbor_event#7

Reply:
Personally I'm more than okay with just erroring here as mixing floats and crypto-sensitive information (i.e. the main use-case for `--preserve-encodings=true`) sounds like a bad idea in the first place.